File: java/src/main/java/example/micronaut/HelloController.java
package example.micronaut;

import io.micronaut.context.annotation.ConfigurationProperties;
import io.micronaut.context.annotation.Value;

import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;

import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Produces;

@Controller("${hello.controller.path:/}") // <1>
public class HelloController {

    private final String someValue;
    public HelloController(@Value("${hello.value}") String someValue) { this.someValue = someValue; }

    @Get // <2>
    public String hello() { return someValue; }

}

File: kotlin/src/main/kotlin/example/micronaut/HelloController.kt
package example.micronaut

import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Produces

import io.micronaut.context.annotation.Value


@Controller("\${hello.controller.path}") // <1>
class HelloController (@Value("\${hello.value}") val secretValue: String) {
    @Get// <2>
    @Produces(MediaType.TEXT_PLAIN) // <3>
    fun index(): String {
        return secretValue
    }
}
